### PR TITLE
Use ECAL ratio timing algorithm for Run 1 and Run 2, and CC timing algorithm for Run 3 and beyond

### PR DESCRIPTION
**Pull request by thomreis**
#### PR description:

This PR sets the default ECAL timing algorithm to the ratio method. This is the default timing algorithm for Run 1 and Run 2.
For Run 3 and Phase 2 a new modifier `run3_ecal` is used to change the timing algorithm to CC timing and use different records with the label 'CC' for the timing calibrations and timing offset constants.

Note that this behaviour is different than what was discussed in the [meeting](https://indico.cern.ch/event/1329876/) on the 27th Sept. when the plan was made to make the CC timing the default and use eras to modify Run 1 and Run 2 configurations to use the ratio method. Since Run 1 configurations could not be modified using eras we decided to reverse the behaviour as in the description above.

#### PR validation:

The PR passes Run 1 and Run2 WFs in the limited matrix tests but currently fails all Run 3 and Phase 2 WFs because the consumed records with the 'CC' label are not yet in the GTs.

### CODE REVIEW

**Comment by saumyaphor4252** [comment]
@cmsbuild, please test

**Comment by thomreis** [comment]
As mentioned in the description exceptions of the `NoProductResolverException` are expected for this PR at the moment because the records are not yet in the GTs. @saumyaphor4252 if you are happy with the current result can we proceed and add them?

**Comment by saumyaphor4252** [comment]
test parameters:
- pull_request=https://github.com/cms-sw/cmssw/pull/42958

**Comment by saumyaphor4252** [comment]
@cmsbuild, please test

File: RecoLocalCalo/EcalRecProducers/python/ecalMultiFitUncalibRecHit_cfi.py [hunk 1]
```diff
@@ -3,3 +3,17 @@
...
+import FWCore.ParameterSet.Config as cms
+from Configuration.Eras.Modifier_run3_ecal_cff import run3_ecal
+run3_ecal.toModify(ecalMultiFitUncalibRecHit,
+    algoPSet = dict(timealgo = 'crossCorrelationMethod',
+        outOfTimeThresholdGain12pEB = 2.5,
+        outOfTimeThresholdGain12mEB = 2.5,
+        outOfTimeThresholdGain61pEB = 2.5,
+        outOfTimeThresholdGain61mEB = 2.5,
+        timeCalibTag = cms.ESInputTag('', 'CC'),
+        timeOffsetTag = cms.ESInputTag('', 'CC')
```
Comment (mmusich) [inline RecoLocalCalo/EcalRecProducers/python/ecalMultiFitUncalibRecHit_cfi.py:16]:
```suggestion
        timeCalibTag = ':CC',
        timeOffsetTag = ':CC'
```

**Review by mmusich — COMMENTED** [review]
just a minor suggestion, you can actually drop cms type specifications for extra safety.

File: RecoLocalCalo/EcalRecProducers/python/ecalMultiFitUncalibRecHit_cfi.py (hunk 1 above)
Comment (thomreis) [reply RecoLocalCalo/EcalRecProducers/python/ecalMultiFitUncalibRecHit_cfi.py:16]: Thanks. I was not quite sure how to handle ESInputTags for this.

**Comment by thomreis** [comment]
Right, this PR actually needs #42932 as well to pass the tests. And that PR needs an extension to explicitly get the right records for the ratio method calibrations since the online DQM configuration usese the `Run3` era modifier.

**Comment by thomreis** [comment]
test parameters:

- pull_requests=#42932,#42958

**Comment by thomreis** [comment]
please test

**Comment by thomreis** [comment]
It seems that `132X_dataRun3_v1` needs to be updated to have the CC labelled records as well in order to pass WF 139.001.

**Comment by mmusich** [comment]
> It seems that 132X_dataRun3_v1 needs to be updated to have the CC labelled records as well in order to pass WF 139.001.

already reported: https://github.com/cms-sw/cmssw/pull/42958#issuecomment-1750216386 (and fixed).

**Comment by thomreis** [comment]
please test

**Comment by saumyaphor4252** [comment]
please test
- Should work now, fixed a missing comma

**Comment by thomreis** [comment]
We expect changes in the ECAL timing with this PR since before many workflows were using the default CC timing algorithm but with the ratio timing calibrations in the GTs. This should be fixed now with this PR.

**Comment by perrotta** [comment]
urgent
- ping @cms-sw/reconstruction-l2 @cms-sw/orp-l2 (for @cms-sw/operations-l2 )

**Comment by thomreis** [comment]
Yes it is possible that the changed timing has small effects on other distributions as well. Depending, e.g. how the out of time flag is set.

**Comment by mandrenguyen** [comment]
+reconstruction
I see a change to quantities related to ECAL time, as intended, and then small changes to other ECAL related quantities which are also expected, as indicated [here](https://github.com/cms-sw/cmssw/pull/42928#issuecomment-1755278886).

**Pull request merged**